src/ratbag_emu/server/device_controller.py

```diff
@@ -18,20 +18,18 @@
 Devices
 '''
 def list_devices():
-    devices = []
+    device_list = []
 
-    for device_id, device in DeviceHandler.devices.items():
+    DeviceHandler.lock.acquire()
+    devices = DeviceHandler.devices.copy()
```

Comment:
not entirely sure if you need to copy it, when you can just release the lock after the for loop.

Reply:
Because `dispatch` will take its time, and it's running on a loop, meaning it will always be running. Copying the device list makes the server much more responsive when we want to modify the device list.